content/articles/language-integrated-query-in-c#/index.md.md

```diff
@@ -0,0 +1,156 @@
+        }
+    }
+}
+```
+### Identifying the difference between an eager and a sluggish evaluation.
+Lazy evaluation means that each call to the iterator only processes a single element from the source collection. Depending on the circumstances, an iterator can be a custom class, a foreach, or a while loop.
+```c#
+public static class Fibonacci
+{
+    public static IEnumerable<long> Series()
+    {
+        long prevX1 = 1;
+        long prevX2 = 1;
+
+        yield return prevX1;
+
+        yield return prevX2;
+
+        while(true)
+        {
+            long temp = prevX1 + prevX2;
+            prevX1 = prevX2;
+            prevX2 = temp;
+            yield return temp;
+        }
+    }
+}
+```
+Eager evaluation is the evaluation method used by most conventional programming languages.
+```c#
+IEnumerable<int> myMethod()
+{
+    for(int i=0; i <=900; i++)
+    {
+        yield return I;
+    }
+}
+...
+
+var ex = myMethod.Take(2);
+var list = ex.ToList();
+```
+The yield keyword allows the collection to be evaluated gradually. Eager evaluation is the process of forcing the entire collection into memory. When you call ToList, ToDictionary, or ToArray, the enumeration is evaluated immediately, and all of the elements are returned in a collection.
+
+### Identifying places in your code where LINQ queries may have performance difficulties
+The sample that we have created above performs an out shuffle, with the top and bottom cards remaining unchanged on every run. Let's make one alteration. instead of an out shuffle, we'll employ an in shuffle, in which all 52 cards shift positions. As a result, the top half's final card becomes the bottom half's final card. This is a straightforward modification to a single line of code. Switch the Take and Skip locations in the current shuffle query to update it. The top and bottom half of the deck will now be in reverse order.
+```c#
+shuffle = shuffle.Skip(26).InterleaveSequenceWith(shuffle.Take(26));
+```
+When you execute the program once more, you will note that the deck 'takes 52 iterations to reshuffle itself. This is contributed to by several factors. You may address one of the key causes of this reduction in performance which is the inefficient lazy evaluation. Note that we used a LINQ query to create the original deck. The three queries of LINQ on the previous deck are used to generate each shuffle. All of this occurs at a glacial speed. This indicates that the sequence is repeated each time it is requested. By the moment you reach the fifty second iteration, you will have regenerated the initial deck severaly i.e more than once. Let's make a log to show how this works. Then you'll take care of it.
+
+Type or copy the method below into your Extensions.cs file. This extension method results into the creation of a new file which is named as debug.log in your project directory and logs the query that is presently being run. This extension method can be used to indicate the completion of a query.

```

Comment:
change to "Type or copy the method below into your `Extensions.cs` file. This extension method creates a new file named.log in your project directory and logs the query that is presently being running. We can use this extension method to indicate the completion of a query."